### PR TITLE
initial commit - cashfree integration

### PR DESCRIPTION
**Pull request by murtraja**
@yusuf4u52 , don't merge as of now, I have to test certain things first. Please start the review though.

### CODE REVIEW

File: users/return.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    echoSuccessOrFailurePage(null, $orderId);
+  }
+} else {
+  die("Wrong protocol");
+}  
+
+
+
+/*
+successful payment scenario
```
Comment (yusuf4u52) [inline users/return.php:123]: Remove all these comments

File: users/checkout.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        'message' => string 'order_expiry_time : Expiry time should be more than 15 min and less than 30 days' (length=80)
+        'code' => string 'order_expiry_time_invalid' (length=25)
+        'type' => string 'invalid_request_error' (length=21)
+    */
+    $message = "Something went wrong with order creation server side! \n";
+    echo $message;
+    $server_message = $resp["data"]["message"];
+    $server_error_code = $resp["data"]["code"];
+    $server_error_type = $resp["data"]["type"];
+    var_dump($resp);
```
Comment (yusuf4u52) [inline users/checkout.php:65]: This seems to be left after debug:)

File: users/common.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+{
+    $curl = curl_init($url);
+
+    curl_setopt($curl, CURLOPT_VERBOSE, 1);
+    curl_setopt($curl, CURLOPT_URL, $url);
+    curl_setopt($curl, CURLOPT_POST, true);
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
+    curl_setopt($curl, CURLOPT_POSTFIELDS, json_encode($data));
+    // remove below properties in production
```
Comment (yusuf4u52) [inline users/common.php:13]: Remove in production :)

File: users/common.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    return array("code" => $httpCode, "data" => $responseJson);
+}
+
+function doGet($url, $headers)
+{
+    $curl = curl_init($url);
+
+    curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+    // remove below properties in production
```
Comment (yusuf4u52) [inline users/common.php:34]: Same here.

File: users/download_receipt_pdf.php [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    }
+    $orderId = $_GET['order_id'];
+    $sql = "SELECT r.* FROM `order_details` od inner join receipts r on od.receipt_no=r.Receipt_No where od.gw_order_id='$orderId'";
+    $result= mysqli_query($link,$sql);
+    $row = mysqli_fetch_assoc($result);
+    if(!$row) {
+        die("Couldn't find the receipt details for this order");
+    }
+    $receiptTemplate = file_get_contents("receipt.html");
+    $pdfContent =  getReceiptHtml($link,$receiptTemplate, $row);
```
Comment (yusuf4u52) [inline users/download_receipt_pdf.php:23]: variable name $htmlContent

File: users/generate_receipt_pdf.php [hunk 6]
```diff
@@ -14,15 +15,6 @@
...
-$dompdf->render();
-// $dompdf->stream("",array("Attachment" => false));
+generate_pdf($pdfContent);
 
-$dompdf->stream("receipts.pdf");
-// $output = $dompdf->output();
-// file_put_contents('Brochure.pdf', $output);
-exit(0);
\ No newline at end of file
+exit(0);
```
Comment (yusuf4u52) [inline users/generate_receipt_pdf.php:20]: What is the purpose of exit here?

**Review by yusuf4u52 — CHANGES_REQUESTED** [review]
See comments in the file

File: users/checkout.php (hunk 2 above)
Comment (murtraja) [reply users/checkout.php:65]: Instead of echoing out to user, logging this internally

**Comment by murtraja** [comment]
@yusuf4u52  tested a few scenarios. Cashfree states in their documentation that it calls notify.php in error condition as well. But while testing, cashfree didn't call notify.php, so couldn't test the error scenario.

**Pull request merged**